crates/bounded_planet_core/src/bin/client/main.rs

```diff
@@ -101,3 +136,207 @@ fn respond_to_pings(
         }
     }
 }
+
+#[derive(Default)]
+pub struct TileReceivedState {
+    pub event_reader: EventReader<ReceiveEvent>,
+}
+
+/// When a tile is received from the server, we load it into the scene
+fn handle_tile_received(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut state: ResMut<TileReceivedState>,
+    receiver: ResMut<Events<ReceiveEvent>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut textures: ResMut<Assets<Texture>>,
+    mut materials: ResMut<Assets<StandardMaterial>>
+) {
+    for evt in state.event_reader.iter(&receiver) {
+        if let ReceiveEvent::ReceivedPacket { connection: ref _connection, data } = evt {
+            if let Packet::WorldTileData(WorldTileData { mesh_data }) = (**data).clone() {
+                info!("Loading tile received from server.");
+                let land_texture_top_handle = asset_server
+                    .load_sync(&mut textures, "content/textures/CoveWorldTop.png")
+                    .expect("Failed to load CoveWorldTop.png");
+                commands.spawn(PbrComponents {
+                    mesh: meshes.add(Mesh {
+                        primitive_topology: bevy::render::pipeline::PrimitiveTopology::TriangleList,
+                        attributes: vec![
+                            VertexAttribute::position(mesh_data.vertices),
+                            VertexAttribute::normal(mesh_data.normals),
+                            VertexAttribute::uv(mesh_data.uvs),
+                        ],
+                        indices: Some(mesh_data.indices),
+                    }),
+                    material: materials.add(StandardMaterial {
+                        albedo_texture: Some(land_texture_top_handle),
+                        shaded: true,
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                });
+                info!("Finished loading tile.");
+            }
+        }
+    }
+}
+
+#[derive(Default)]
```

Comment:
what is this?

Reply:
As in, add some docs regarding it.

Reply:
`RequestTileOnConnected` was incorrectly named and now is `RequestTileOnConnectedState`